### DISCUSSION TITLE
Running Pluto without Webbserver.jl?

### DISCUSSION THREAD

**Discussion by zahachtah** [fonsp/Pluto.jl#2184] [Pluto development] [answered]
I love Pluto, it's become my one and only way to write Julia code. 

Out of curiosity, and with a possible useful implementation in mind (interacting with notebooks via a remote database), would it be possible to use Pluto without the webserver and pluto-frontend? If I had another Julia process, could I send cell-changes directly and Retrieve the results without going over websockets/http? Where would I best interface the code to achieve this? 

process_ws_message seems like one function one could "hijack" for incoming messages, but where would I find updates after code is evaluated? is that in the client channels? Which function makes sure to send them back? I have been digging but Pluto has a biiig codebase now!

I'd like to keep as much as possible of Pluto functionality, its a fantastic piece of work, I'd just like to see if it can be used in a different way. Any hints would be most welcome!

**Comment by Pangoraw** [comment, marked as answer]
You can try to control notebooks using the internal Julia API. See some of the tests for an example:

https://github.com/fonsp/Pluto.jl/blob/4758c4e73e472808b1dad1b765fc12bfb3cd4071/test/React.jl#L16-L40

**Reply by zahachtah** [reply]
Perfect, thanks!
